src/emailcontactmodel.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2020- 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "emailcontactmodel.h"
+#include <QHash>
+
+
+namespace  {
+    const int MAX_COUNT = 100;
+}
+
+
+EmailContactModel::EmailContactModel(QAbstractItemModel *parent)
+    : QAbstractListModel(parent)
+{
+    initialize(MAX_COUNT);
+}
+
+
+static void push(QHash<QMailAddress, QMailTimeStamp> &container, const QMailAddress &address, const QMailTimeStamp &stamp)
```

Comment:
Small thing but commonly nicer to use pointers instead of non-const references. Even if more obvious with this method, you don't commonly see from foo(bar) that it can have side-effects on the passed parameter.

Reply:
done